### PR TITLE
Raise error when function call has wrong number of arguments

### PR DESCRIPTION
**Pull request by EmilyBourne**
Pyccel is not guaranteed to work when the underlying python code will not run, however the error raised when too few arguments are passed to the function is particularly unclear:
```
ERROR at code generation stage
pyccel:
 |fatal [codegen]: tmp.py| _print_NoneType is not yet implemented for language : Fortran
Pyccel has encountered syntax that has not been implemented yet. Please create an issue at https://github.com/pyccel/pyccel/issues and provide a small example of your problem. Do not forget to specify your target language
```
The `NoneType` appears as pyccel uses `None` internally as a placeholder for arguments which have no default value. It is trivial to test that the number of arguments is correct and to raise a more human readable error with an identifiable line of code

**Commit Summary**
- Stop obfuscation of fst when one pyccel object is represented by multiple python.ast nodes (e.g. `pyccel.ast.Assign` is represented by `ast.assign` and `ast.Expr`)
- Raise an error if too many arguments are passed to a function
- Raise an error if too few arguments are passed to a function

### CODE REVIEW

File: pyccel/parser/semantic.py [hunk 1]
```diff
@@ -902,7 +902,7 @@ def _handle_function_args(self, arguments, **settings):
                 args.append(a)
         return args
 
-    def _handle_function(self, func, args, **settings):
+    def _handle_function(self, expr, func, args, **settings):
         if not isinstance(func, (FunctionDef, Interface)):
             args, kwargs = split_positional_keyword_arguments(*args)
             for a in args:
```
Comment (yguclu) [inline pyccel/parser/semantic.py:908]: Could you please add a docstring to this function? It has become really obscure.

File: pyccel/parser/semantic.py [hunk 2]
```diff
@@ -915,8 +915,16 @@ def _handle_function(self, func, args, **settings):
 
             return expr
```
Comment (yguclu) [inline pyccel/parser/semantic.py:916]: Shouldn't we call this variable `new_expr` as done below?

**Review by yguclu — CHANGES_REQUESTED** [review]

File: pyccel/parser/semantic.py (hunk 2 above)
Comment (EmilyBourne) [reply pyccel/parser/semantic.py]: Fixed in 008e068

File: pyccel/parser/semantic.py (hunk 1 above)
Comment (EmilyBourne) [reply pyccel/parser/semantic.py]: Fixed in 008e068

File: pyccel/parser/semantic.py [hunk 3]
```diff
@@ -903,6 +903,19 @@ def _handle_function_args(self, arguments, **settings):
         return args
 
     def _handle_function(self, expr, func, args, **settings):
+        """
+        Create a FunctionCall or an instance of a PyccelInternalFunction
+        from the function informations and arguments
```
Comment (yguclu) [inline pyccel/parser/semantic.py:908]: Isn't "information" an uncountable noun in English?

File: pyccel/parser/semantic.py [hunk 4]
```diff
@@ -902,7 +902,20 @@ def _handle_function_args(self, arguments, **settings):
...
+
+        Parameters
+        ==========
+        expr : PyccelAstNode
+               The expression where this call is found (used for error output)
+        func : FunctionDef, Interface or PyccelInternalFunction type
+               The function being called
+        args : tuple
+               The arguments passed to the function
+        """
```
Comment (yguclu) [inline pyccel/parser/semantic.py:918]: Thanks! But I think the "Returns" section is missing...

File: pyccel/parser/semantic.py (hunk 3 above)
Comment (EmilyBourne) [reply pyccel/parser/semantic.py]: Fixed in 47de259

File: pyccel/parser/semantic.py (hunk 4 above)
Comment (EmilyBourne) [reply pyccel/parser/semantic.py]: Fixed in 47de259

**Review by yguclu — APPROVED** [review]

**Pull request merged**